cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -137,9 +146,15 @@ func (m *azureCache) regenerate() error {
 		}
 	}
 
+	managedPoolNames := sets.New[string]()
 	// Regenerate VMSS to autoscaling options mapping.
 	newAutoscalingOptions := make(map[azureRef]map[string]string)
 	for _, vmss := range m.scaleSets {
+		// Check if the nodegroup is registered. If it is, lets store the managedPoolName
+		managedPoolName := extractAKSManagedPoolNameFromTags(vmss.Tags)
+		if registeredNodeGroupIDs.Has(managedPoolName) && managedPoolName != "" {
```

Comment:
need to check this logic a bit closer for registeredNodeGroupIDs and do some manual testing.